### PR TITLE
[TS] Migrate Notification tests to .tsx

### PR DESCRIPTION
**Pull request by nishantdecode**
What does this PR do?
This PR migrates MaskedInput component tests from *.js to *.tsx to allow ts-jest to catch type-related errors in the unit testing pipeline. Changes made:

1. Removed truncation from custom theme as they are not defined in base.js
2. Removed children prop for Notification.
3. Added id prop to declaration file for notification

Where should the reviewer start?
Ensure all tests are currently passing

What testing has been done on this PR?
Manually ran and passed MaskedInput component tests locally.
CircleCI tests have passed.

How should this be manually tested?
Modified tests behave identically as before.

Do Jest tests follow these best practices?

Any background context you want to provide?

What are the relevant issues?
#5599 

Screenshots (if appropriate)

Do the grommet docs need to be updated?
no

Should this PR be mentioned in the release notes?
no

Is this change backwards compatible or is it a breaking change?
compatible

### CODE REVIEW

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 1]
```diff
@@ -63,30 +64,28 @@ describe('Notification', () => {
...
   });
 
-  [
+  const validPositions: LayerPositionType[] = [
     'top',
     'bottom',
     'left',
     'right',
-    'start',
-    'end',
```
Comment (nishantdecode) [inline src/js/components/Notification/__tests__/Notification-test.tsx:72]: Removed 'start' and 'end' as they are not valid string literals within the defined type  LayerPositionType.

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 2]
```diff
@@ -63,30 +64,28 @@ describe('Notification', () => {
...
         <Grommet>
           <Notification
             id="position-test"
-            toast={{ position: positions }}
+            toast={{ position }}
             title="title"
-            message="message"
-          >
-            This is a layer
-          </Notification>
```
Comment (nishantdecode) [inline src/js/components/Notification/__tests__/Notification-test.tsx:89]: I am not sure if this is the intended use case of Notification, as no children props are being destructured and used.

File: src/js/components/Notification/__tests__/Notification-test.tsx (hunk 1 above)
Comment (jcfilben) [reply src/js/components/Notification/__tests__/Notification-test.tsx:72]: I think the LayerPositionType should be change to include 'start' and 'end'

File: src/js/components/Notification/__tests__/Notification-test.tsx (hunk 2 above)
Comment (jcfilben) [reply src/js/components/Notification/__tests__/Notification-test.tsx:89]: It looks like Notification currently isn't expecting anything to be passed through as children. Can we change this test to just remove the `This is a layer` part?

File: src/js/components/Notification/__tests__/Notification-test.tsx (hunk 1 above)
Comment (nishantdecode) [reply src/js/components/Notification/__tests__/Notification-test.tsx:72]: yes I will include it.

File: src/js/components/Notification/__tests__/Notification-test.tsx (hunk 2 above)
Comment (nishantdecode) [reply src/js/components/Notification/__tests__/Notification-test.tsx:89]: yes I will change it to:

```javascript
<Notification
    id="position-test"
    toast={{ position }}
    title="title"
    message="message"
/>
```

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 3]
```diff
@@ -75,18 +76,18 @@ describe('Notification', () => {
     'top-right',
     'bottom-left',
     'bottom-right',
-  ].forEach((positions) =>
-    test(`position ${positions}`, () => {
+  ];
+
+  validPositions.forEach((position) =>
+    test(`position ${position}`, () => {
```
Comment (jcfilben) [inline src/js/components/Notification/__tests__/Notification-test.tsx:82]:
```suggestion
  validPositions.forEach((positions) =>
    test(`position ${positions}`, () => {
```

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 4]
```diff
@@ -75,18 +76,18 @@ describe('Notification', () => {
...
       render(
         <Grommet>
           <Notification
             id="position-test"
-            toast={{ position: positions }}
+            toast={{ position }}
```
Comment (jcfilben) [inline src/js/components/Notification/__tests__/Notification-test.tsx:87]:
```suggestion
            toast={{ position: positions }}
```

File: src/js/components/Notification/__tests__/Notification-test.tsx [hunk 5]
```diff
@@ -218,16 +219,14 @@ describe('Notification', () => {
...
+    const theme: ThemeType = {
       notification: {
         direction: 'row',
-        truncation: true,
         container: {
           pad: 'medium',
         },
         toast: {
           direction: 'column',
-          truncation: false,
```
Comment (jcfilben) [inline src/js/components/Notification/__tests__/Notification-test.tsx:230]: Can we keep the `truncation: true` and `truncation: false`?

Comment (nishantdecode) [reply src/js/components/Notification/__tests__/Notification-test.tsx:230]: Yes we can but the necessary changes are need to be made in base.d.ts

Comment (nishantdecode) [reply src/js/components/Notification/__tests__/Notification-test.tsx:230]: but base.js does not have any truncation property for notification or notification.toast

Comment (jcfilben) [reply src/js/components/Notification/__tests__/Notification-test.tsx:230]: Ah gotcha, looks like we don't document this prop either. In that case I'm fine removing `truncation` from this test

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by MikeKingdom — APPROVED** [review]

**Pull request merged**